delayrecorder.py

```diff
@@ -63,115 +64,118 @@ def parse_arguments():
-                )
+
+    args = parser.parse_args()
+    if args.filename.suffix != ".csv":
+        print("Error: Provided filename is invalid or does not have .csv extension")
+        sys.exit(1)
+    return args
+
+
+def find_arduino_on_serial_port() -> serial.Serial:
+    devices = serial.tools.list_ports.comports()
+    for device in devices:
+        if device.manufacturer is not None:
+            if "Arduino" in device.manufacturer:
+                print(f"Found Arduino at {device[0]}")
+                return serial.Serial(device[0], 115200, timeout=5)
+
+    raise ConnectionRefusedError("Did not find Arduino on any serial port. Is it connected?")
+
+
+def read_measurements_from_arduino(num_measurements: int, quiet_mode: bool) -> List[float]:
+    try:
+        ser = find_arduino_on_serial_port()
+
+        print(f"Collecting {num_measurements} measurements from the Arduino")
+        if quiet_mode:
+            print("Running in quiet mode, won't print the measurements to the terminal")
+
+        # Read messages from Arduino
+        timeout = time.time() + 0.01
+        while True:
+            a = ser.readline().decode()
+            if time.time() > timeout:
+                break
+
+        measurements = []
+        i = 0
+        overall_rounds = 0
+        init_message = 0
+
+        while i < num_measurements:
+            overall_rounds += 1
+            a = ser.readline().decode()
+            if "." in a:
+                init_message = 1
+                i += 1
+                a = a.replace("\n", "")
+                a = a.replace("\r", "")
+                measurements.append(float(a))
+                if not quiet_mode:
+                    print(f"G2G Delay trial {i}/{num_measurements}: {a} ms")
+                else:
+                    print(f"G2G Delay trial {i}/{num_measurements}", end="\r")
             else:
-                print(
-                    """Did not receive msmt data from the Arduino for 5 seconds.
+                if overall_rounds > 0 and init_message == 1:
+                    print(
+                        "Did not receive msmt data from the Arduino for another 5 seconds. "
+                        "Is the phototransistor still sensing the LED?"
+                    )
+                else:
+                    print(
+                        """Did not receive msmt data from the Arduino for 5 seconds.
 Is the phototransistor sensing the LED on the screen?
 Is the correct side of the PT pointing towards the screen (the flat side with the knob on it)?
 Is the screen brightness high enough (max recommended)?"""
-                )
-                init_message = 1
+                    )
+                    init_message = 1
+    except ConnectionRefusedError as e:
+        print(f"ConnectionRefusedError error. {e}")
+        sys.exit(1)
+    except serial.SerialException:
+        print("SerialException error. Did you unplug the Arduino?")
+        sys.exit(1)
```

Comment:
Printing the exception message and exiting with an error is exactly what the exception does by itself if you don't catch it. I therefore suggest reverting your last commit. Agreed?

Reply:
Can you describe the expected behaviour for when it can't find the arduino and for when someone unplugs it? I'll take a look again on monday

Reply:
The program shall throw a `serial.SerialException` with a sufficiently expressive error message. I expect that users will have no difficulties assigning the `serial.SerialException` to their action when they unplugged the Arduino or it somehow lost contact.

Reply:
Additional benefit of letting the exception handle itself: it displays the call stack, so you know at which point during runtime the exception was raised.

Reply:
Okay, I think I understand your reasoning now and have pushed a revert commit. Feel free to rebase to get rid of the commits altogether, I'll leave that one up to you.

Reply:
Cool, thank you! I'll squash and merge 👍